### ISSUE TITLE
Error when tapping on  "start playing in the background" on PeerTube 

### ISSUE THREAD

**Issue by nicoursi** [TeamNewPipe/NewPipe#8660]
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.23.1

### Steps to reproduce the bug

1. Search for any video on PeerTube
1. Long press on the video you like
2. Tap on "start playing in the background"

### Expected behavior

The stream should start playing. If not possible to play in the background, the "play in the background" option should not be available from the list.

### Actual behavior

The stream does not start and a bug report notification appears. 

### Screenshots/Screen recordings

<img alt="Screenshot_20220719-170319483 (1)" src="https://user-images.githubusercontent.com/16888339/179783535-94cfa721-5c58-4f38-9f79-8bd43039aa4d.jpg" height=600 />

## Exception
* __User Action:__ play stream
* __Request:__ Loading failed for [Cat cat cat cat cat]: https://diode.zone/videos/watch/a18d2f95-74a3-4a90-a157-8938e922fa77
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ PeerTube
* __Version:__ 0.23.1
* __OS:__ Linux Android 11 - 30
<details><summary><b>Crash log </b></summary><p>

```
org.schabi.newpipe.player.mediasource.FailedMediaSource$MediaSourceResolutionException: Unable to resolve source from stream info. URL: https://diode.zone/videos/watch/a18d2f95-74a3-4a90-a157-8938e922fa77, audio count: 0, video count: 0, 12
	at org.schabi.newpipe.player.playback.MediaSourceManager.lambda$getLoadedMediaSource$3(MediaSourceManager.java:432)
	at org.schabi.newpipe.player.playback.MediaSourceManager.$r8$lambda$GiL2pCn9PgSZNVk872SsYVyH27I(Unknown Source:0)
	at org.schabi.newpipe.player.playback.MediaSourceManager$$ExternalSyntheticLambda2.apply(Unknown Source:6)
	at io.reactivex.rxjava3.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:58)
	at io.reactivex.rxjava3.internal.operators.single.SingleDoOnError$DoOnError.onSuccess(SingleDoOnError.java:52)
	at io.reactivex.rxjava3.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.onSuccess(SingleSubscribeOn.java:68)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeToSingle$ToSingleMaybeSubscriber.onSuccess(MaybeToSingle.java:83)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onNext(FlowableElementAtMaybe.java:80)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:136)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:15868)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.rxjava3.core.Maybe.subscribe(Maybe.java:5330)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:644)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)

```
</details>
<hr>

### Affected Android/Custom ROM version

Android 11 / LineageOS 18.1

### Affected device model

Redmi Note 7

### Additional information

This bug is very annoying when trying to play in the background some videos from "Whats New" as you can't tell in advance if those videos are from YouTube or PeerTube.

**Comment by ShareASmile** [comment]
Fixed in [v0.25.0](https://github.com/TeamNewPipe/NewPipe/issues/9712) by #9562

**Comment by ShareASmile** [comment]
Sorry, I was busy lately, will reply soon.
